project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_visitors/c4m_og_invite_visitors.module

```diff
@@ -0,0 +1,395 @@
+    $form_state['account'][] = $account;
+
+    // Invoke hook_c4m_og_invite_visitors_invited() - after a user is created
+    // and added to group.
+    module_invoke_all('c4m_og_invite_visitors_invited', $account, $og_membership);
+
+    // Print the message for each user.
+    drupal_set_message($message);
+  }
+}
+
+/**
+ * Implements hook_mail().
+ */
+function c4m_og_invite_visitors_mail($key, &$message, $params) {
+  $language = $message['language'];
+  $variables = array(
+    'user' => $params['account'],
+    'message' => $message,
+    'form_fields' => $params['form_fields'],
+  );
+  $message['subject'] .= _c4m_og_invite_visitors_mail_text($key . '_subject', $language, $variables);
+  $message['body'][] = _c4m_og_invite_visitors_mail_text($key . '_body', $language, $variables);
+}
+
+/**
+ * Returns a mail string for a variable name.
+ *
+ * Used by c4m_og_invite_visitors_mail() and the settings forms to retrieve
+ * strings.
+ */
+function _c4m_og_invite_visitors_mail_text($key, $language = NULL, $variables = array(), $replace = TRUE) {
+  global $language;
+
+  // Try to load the text first from variable.
+  $text = variable_get($key, FALSE);
+
+  if ($replace) {
+    // We do not sanitize the token replacement, since the output of this
+    // replacement is intended for an e-mail message, not a web browser.
+    return token_replace($text, $variables, array(
+      'language' => $language,
+      'callback' => 'c4m_og_invite_visitors_mail_tokens',
+      'sanitize' => FALSE,
+      'clear' => TRUE,
+    )
+    );
+  }
+
+  return $text;
+}
+
+/**
+ * Token callback to add unsafe tokens for user mails.
+ *
+ * This function is used by the token_replace() call at the end of
+ * _c4m_og_invite_visitors_mail_text() to set up some additional tokens that can
+ * be used in email messages generated by c4m_og_invite_visitors_mail().
+ *
+ * @param array $replacements
+ *   An associative array variable containing mappings from token names to
+ *   values (for use with strtr()).
+ * @param array $data
+ *   An associative array of token replacement values. If the 'user' element
+ *   exists, it must contain a user account object with the following
+ *   properties:
+ *   - login: The UNIX timestamp of the user's last login.
+ *   - pass: The hashed account login password.
+ * @param array $options
+ *   Unused parameter required by the token_replace() function.
+ */
+function c4m_og_invite_visitors_mail_tokens(&$replacements, $data, $options) {
+  global $user, $base_url;
+  if (empty($data['user'])) {
+    return;
+  }
+  $group = c4m_og_current_group();
+  $url_options = array(
+    'absolute' => TRUE,
+    'purl' => array(
+      'provider' => 'og_purl|node',
+      'id' => $group->nid,
+    ),
+  );
+
+  $replacements['[user:full_name]'] = c4m_user_name($user);
+  $replacements['[user:public-profile-url]'] = $base_url . '/users/' . preg_replace('@[^a-z0-9-]+@', '-', strtolower(trim($replacements['[user:full_name]'])));
```

Comment:
@ybaras 
I see that we're using the same `preg_replace('@[^a-z0-9-]+@'` more than once,
it is better taking it out to an external function (also for a better documentation).